src/components/Button/Button.tsx

```diff
@@ -196,9 +194,7 @@ class Button extends UIComponent<Extendable<IButtonProps>, any> {
   }
 
   private handleFocus = (e: React.SyntheticEvent) => {
-    const isFromKeyboard = whatInput.ask() === 'keyboard'
-
-    this.setState({ isFromKeyboard })
+    this.setState(isFromKeyboard.state())
 
```

Comment:
Are these changes suppose to be part of this PR?

Reply:
I will make a super small PR to separate this. not necessarily part of this one, but as I was using the same code as it's used in button, I just moved it to a separate function